src/scheduler.cpp

```diff
@@ -71,10 +49,11 @@ void CScheduler::serviceQueue()
             taskQueue.erase(taskQueue.begin());
 
             {
-                // Unlock before calling f, so it can reschedule itself or another task
+                // Unlock while calling f, so it can reschedule itself or another task
                 // without deadlocking:
-                reverse_lock<boost::unique_lock<boost::mutex> > rlock(lock);
+                lock.unlock();
                 f();
+                lock.lock();
```

Comment:
You make a good point, I totally missed the exception safety as a feature of reverse_lock. Added it back.